### PR TITLE
Adding hamming numbers algorithm

### LINKED ISSUE #1027: Hamming numbers in c,c++,js with readme.md
Can i start working on it ??

### PR DESCRIPTION
**Pull request by khushal87**
Fixes #1027  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added hamming_numbers algorithm in c,c++,js 
- Added readme.md for hamming numbers

#### Languages Used:
- c
- c++
- JS

#### Files Added:
-  Hamming_Numbers/Hamming_Numbers.c
-  Hamming_Numbers/Hamming_Numbers.cpp
-  Hamming_Numbers/Hamming_Numbers.js
- readme.md

### CODE REVIEW

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//Function for checking whether the number is hamming number or not
+bool is_hamming (int n) {
+    if (n == 1)
+    {
+        return true;
+    }
+    if (n%2==0 || n%3==0 || n%5==0) {
+	   for (int i = 2; i*i < n; ++i) {
+	    if (n%i == 0) {
+	        if (i%2 != 0 and i%3!=0 and i%5!=0)
```
Comment (KanchanThareja) [inline Hamming_Numbers/Hamming_Numbers.cpp]: add spacing around operators =, %

File: Hamming_Numbers/Hamming_Numbers.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+int is_hamming (int n) {
+    if (n == 1)
+    {
+        return 1;
+    }
+    if (n%2==0 || n%3==0 || n%5==0) {
+        for (int i = 2; i*i < n; ++i) {
+	    if (n%i == 0) {
+		if (i%2 != 0 && i%3!=0 && i%5!=0)       
+		    return 0;
```
Comment (KanchanThareja) [inline Hamming_Numbers/Hamming_Numbers.c]: spaces around operators

**Comment by khushal87** [comment]
Done @KanchanThareja

**Comment by jainaman224** [comment]
@KanchanThareja Please review.

**Comment by KanchanThareja** [comment]
@khushal87  you haven't included an md file while raising the issue.
@jainaman224 is it ok to add a readme file without mentioning in the issue itself?

File: Hamming_Numbers/Hamming_Numbers.c [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+//Function for checking whether the number is hamming number or not
+int is_hamming (int n) {
+    if (n == 1)
+    {
+        return 1;
+    }
+    if (n % 2 == 0 || n % 3 == 0 || n % 5 == 0) {
+        for (int i = 2; i * i < n; ++i) {
+	    if (n % i == 0) {
+		if (i % 2 != 0 && i % 3 !=0 && i % 5 !=0)
```
Comment (KanchanThareja) [inline Hamming_Numbers/Hamming_Numbers.c]: there is 3 space indentation in line13. change

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+//Function for checking whether the number is hamming number or not
+bool is_hamming (int n) {
+    if (n == 1)
+    {
+        return true;
+    }
+    if (n % 2 == 0 || n % 3 == 0 || n % 5 == 0) {
+	   for (int i = 2; i*i < n; ++i) {
+	    if (n % i == 0) {
+	        if (i % 2 != 0 and i % 3 != 0 and i % 5 != 0)
```
Comment (KanchanThareja) [inline Hamming_Numbers/Hamming_Numbers.cpp]: indentation

**Comment by KanchanThareja** [comment]
@khushal87 also try improving on comments for this one.

**Comment by KanchanThareja** [comment]
Don't forget to squash your commits

**Comment by marwahmanbir** [comment]
@khushal87 are you working on this?

**Comment by khushal87** [comment]
@marwahmanbir yes

**Comment by khushal87** [comment]
> 
> 
> @khushal87 you haven't included an md file while raising the issue.
> @jainaman224 is it ok to add a readme file without mentioning in the issue itself?

in that case i am adding it?

**Comment by khushal87** [comment]
Please see to it if there are any more changes.. if not can i squash now for final merge ?

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+using namespace std;
+
+//Function for checking whether the number is hamming number or not
+bool is_hamming (int n) {
+    if (n == 1)
+    {
+        return true;
+    }
+    if (n % 2 == 0 || n % 3 == 0 || n % 5 == 0) {
+        for (int i = 2; i*i < n; ++i) {
```
Comment (MastersAbh) [inline Hamming_Numbers/Hamming_Numbers.cpp]: spaces around *

Comment (khushal87) [reply Hamming_Numbers/Hamming_Numbers.cpp]: done

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            }
+        }
+        return true;
+    }
+    return false;
+}
+//If hamming number is found in above function then it is printed in the main function for the given range
+int main () {
+    int n;
+    cin >> n;           //user input
```
Comment (MastersAbh) [inline Hamming_Numbers/Hamming_Numbers.cpp]: please remove spaces around >>

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    int count = 0;
+    int i = 1;
+    bool first = true;
+    while (count < n) {
+        if (is_hamming(i)) {
+            if (!first)
+                cout << " ";
+            cout << i;
```
Comment (MastersAbh) [inline Hamming_Numbers/Hamming_Numbers.cpp]: please remove spaces around <<

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 8]
```diff
@@ -0,0 +1,48 @@
+//Program for finding first n hamming numbers for user input of n
+#include <iostream>
+#include <cmath>
```
Comment (MastersAbh) [inline Hamming_Numbers/Hamming_Numbers.cpp]: please remove space before <

File: Hamming_Numbers/Hamming_Numbers.c [hunk 9]
```diff
@@ -0,0 +1,47 @@
+//Program for finding first n hamming numbers for user input of n
+#include <stdio.h>
+#include <math.h>
```
Comment (MastersAbh) [inline Hamming_Numbers/Hamming_Numbers.c]: please remove space before <

**Comment by khushal87** [comment]
> 
> 
> Please make these changes. Rest looks good.

done

File: Hamming_Numbers/readme.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+# Hamming Number
```
Comment (marwahmanbir) [inline Hamming_Numbers/readme.md]: Hey, can you add some more content in the README? Algorithm, pseudocode, example, time complexity etc?

**Comment by jainaman224** [comment]
@khushal87 Please complete this as soon as possible.

**Comment by MastersAbh** [comment]
Hi @khushal87 , it would be great if you could add some more content in ReadMe as suggested by @marwahmanbir .

**Comment by khushal87** [comment]
thanks @MastersAbh

File: Hamming_Numbers/Hamming_Numbers.c [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+//Function for checking whether the number is hamming number or not
+int is_hamming(int n) {
+    if (n == 1)
+    {
+        return 1;
+    }
+    if (n % 2 == 0 || n % 3 == 0 || n % 5 == 0) {
+        for (int i = 2; i * i < n; ++i) {
+	    if (n % i == 0) {
```
Comment (marwahmanbir) [inline Hamming_Numbers/Hamming_Numbers.c]: Fix indentation in complete program from this line onwards.

File: Hamming_Numbers/Hamming_Numbers.c [hunk 12]
```diff
@@ -0,0 +1,47 @@
+//Program for finding first n hamming numbers for user input of n
```
Comment (marwahmanbir) [inline Hamming_Numbers/Hamming_Numbers.c]: Add a space after "//" in comments.

File: Hamming_Numbers/Hamming_Numbers.c [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+		if ((n / i) % 2 != 0 && (n / i) % 3 != 0 && (n / i) % 5 != 0) 
+                    return 0;
+	    }
+	}
+        return 1;		
+    }
+    return 0;
+}
+//If hamming number is found in above function then it is printed in the main function for the given range
+int main () {
```
Comment (marwahmanbir) [inline Hamming_Numbers/Hamming_Numbers.c]: Remove space after ``main``.

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 14]
```diff
@@ -0,0 +1,48 @@
+//Program for finding first n hamming numbers for user input of n
```
Comment (marwahmanbir) [inline Hamming_Numbers/Hamming_Numbers.cpp]: Space after "//" in all comments.

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+//Function for checking whether the number is hamming number or not
+bool is_hamming (int n) {
+    if (n == 1)
+    {
+        return true;
+    }
+    if (n % 2 == 0 || n % 3 == 0 || n % 5 == 0) {
+        for (int i = 2; i * i < n; ++i) {
+            if (n % i == 0) {
+                if (i % 2 != 0 and i % 3 != 0 and i % 5 != 0)
```
Comment (marwahmanbir) [inline Hamming_Numbers/Hamming_Numbers.cpp]: Always use ``&&`` in place of "and", and ``||`` in place of "or".
Fix this in all files.

File: Hamming_Numbers/Hamming_Numbers.cpp [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+int main () {
+    int n;
+    cin>>n;           //user input
+    int count = 0;
+    int i = 1;
+    bool first = true;
+    while (count < n) {
+        if (is_hamming(i)) {
+            if (!first)
+                cout<<" ";
```
Comment (marwahmanbir) [inline Hamming_Numbers/Hamming_Numbers.cpp]: Indentation is perfect in this file.

File: Hamming_Numbers/Hamming_Numbers.js [hunk 17]
```diff
@@ -0,0 +1,27 @@
+//Function for finding hamming numbers
```
Comment (marwahmanbir) [inline Hamming_Numbers/Hamming_Numbers.js]: Space after "//" in all comments.
Rest looks good in this file.

**Comment by marwahmanbir** [comment]
It would be great if you can try to improve the README as well. @khushal87

**Comment by marwahmanbir** [comment]
@khushal87 are you working on this?

**Comment by khushal87** [comment]
@marwahmanbir only the readme.md is left to work upon ..

**Comment by MastersAbh** [comment]
Please do it as soon as possible @khushal87 and it is ready to be merged

**Comment by khushal87** [comment]
Sorry but i wont be able to update the readME.md file within the stipulated time. Would the code be merged ..? i will do it soon within some day .. i was busy with a project and some other work @MastersAbh

**Comment by MastersAbh** [comment]
@rishabhgarg25699, @marwahmanbir  please have a look.

**Comment by rishabhgarg25699** [comment]
@khushal87 Commits are also not squashed .

**Comment by rishabhgarg25699** [comment]
@jainaman224 See this .

**Comment by marwahmanbir** [comment]
@khushal87 you didn't even fix the indentation! The code still contains tab indentation, please check my last review.
You still have a few minutes until you can commit the changes to the code. 
We can allow README to be fixed later.

**Comment by marwahmanbir** [comment]
Else, up to @jainaman224 to take the decision if we can consider this PR or not.

**Comment by khushal87** [comment]
@marwahmanbir  all the changes are done and the .cpp file has indentation already fixed according to me ..so there is no changes to be done .. rest all are fixed ..sorry for the delay

**Comment by marwahmanbir** [comment]
Unfortunately, the indentation **still** doesn't seem to be fixed yet in C file.

**Comment by jainaman224** [comment]
@khushal87 Please fix indentation.

**Pull request closed without merging**